File: C++/service/TamanhoService.cpp
#include "TamanhoService.h"

// Construtor
TamanhoService::TamanhoService(TamanhoDao& dao) : tamanhoDao(dao) {}

// Obter Tamanho por ID
Tamanho TamanhoService::getById(int id) {
    return tamanhoDao.findById(id);
}

// Obter todos os Tamanhos
std::vector<Tamanho> TamanhoService::getAll() {
    return tamanhoDao.findAll();
}

// Obter todos os Tamanhos por TipoProduto
std::vector<Tamanho> TamanhoService::getAllByTipoProduto(const std::string& tipoProduto) {
    return tamanhoDao.findAllByTipoProduto(tipoProduto);
}

// Inserir um novo Tamanho
void TamanhoService::insert(const Tamanho& tamanho) {
    tamanhoDao.insert(tamanho);
}

// Atualizar um Tamanho existente
void TamanhoService::update(const Tamanho& tamanho) {
    tamanhoDao.update(tamanho);
}

// Deletar um Tamanho pelo ID
void TamanhoService::deleteById(int id) {
    tamanhoDao.deleteById(id);
}

File: JAVA/src/main/java/com/inf_112/trabalho_final/service/TamanhoService.java
package com.inf_112.trabalho_final.service;

import com.inf_112.trabalho_final.dao.TamanhoDao;
import com.inf_112.trabalho_final.model.Tamanho;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class TamanhoService {

    private final TamanhoDao TamanhoDao;

    public TamanhoService(TamanhoDao TamanhoDao) {
        this.TamanhoDao = TamanhoDao;
    }

    public Tamanho getById(int id) {
        return TamanhoDao.findById(id);
    }

    public List<Tamanho> getAll() {
        return TamanhoDao.findAll();
    }

    public List<Tamanho> getAllByTipoProduto(String tipoProduto) {
        return TamanhoDao.findAllByTipoProduto(tipoProduto);
    }

    public void insert(Tamanho Tamanho) {
        TamanhoDao.insert(Tamanho);
    }

    public void update(Tamanho Tamanho) {
        TamanhoDao.update(Tamanho);
    }

    public void delete(int id) {
        TamanhoDao.delete(id);
    }
}
